dev-tools/scripts/cherrypick.sh

```diff
@@ -0,0 +1,191 @@
+ -a          Enters automated mode. Aborts cherry-pick and exits on error
+ -r <remote> Specify remote to push to. Defaults to if other than 'origin'
+ -p          Push to remote. Only done if both cherry-pick and tests succeeded
+    WARNING: Never push changes to a remote branch before a thorough local test
+
+Simple script for aiding in back-porting one more (trivial) commits to other branches.
+On merge conflict the script will run 'git mergetool'. See 'git mergetool --help'
+for help on configuring your favourite merge tool. Check out Sublime Merge (smerge).
+
+Example:
+  # Backport two commits to both stable and release branch
+  dev-tools/scripts/cherrypick.sh -b branch_9x  deadbeef0000 cafebabe1111
+  dev-tools/scripts/cherrypick.sh -b branch_9_0 deadbeef0000 cafebabe1111
+EOF
+}
+
+function yesno() {
+  question=$1
+  unset answer
+  echo "$question"
+  while [[ "$answer" != "y" ]] && [[ "$answer" != "n" ]]; do
+    read -r answer
+    if [[ "$answer" == "y" ]]; then
+      true
+    else
+      false
+    fi
+  done
+}
+
+GIT_COMMAND=git
+PRECOMMIT="true"
+TESTARG="-x test"
+TEST=
+TESTS_PASSED=
+PUSH=
+REMOTE=origin
+NOPULL=
+AUTO_MODE=
+
+while getopts ":b:phstnar:" opt; do
+  case ${opt} in
+    b)
+      BRANCH=$OPTARG
+      ;;
+    r)
+      REMOTE=$OPTARG
+      ;;
+    p)
+      PUSH=true
+      ;;
+    s)
+      PRECOMMIT=
+      ;;
+    a)
+      AUTO_MODE="true"
+      ;;
+    n)
+      NOPULL="true"
+      ;;
+    t)
+      TEST="true"
+      TESTARG=""
+      ;;
+    h)
+      usage
+      exit 0
+      ;;
+   \?)
+      echo "Unknown option $OPTARG"
+      usage
+      exit 1
+   esac
+done
+shift $((OPTIND -1))
+
+COMMITS=( "$@" )
+
+if [ -z "$BRANCH" ]; then
+  LOG INFO "Lacking -b option, must specify target branch"
+  usage
+  exit
+fi
+
+if [ ${#COMMITS[@]} -eq 0 ]; then
+  LOG ERROR "Please specify one or more commits"
+  usage
+  exit
+fi
+
+LOG INFO "Checking out target branch $BRANCH"
+$GIT_COMMAND checkout "$BRANCH"
+if [[ ! "$NOPULL" ]]; then
+  $GIT_COMMAND pull --ff-only
+else
+  LOG INFO "Skipping git pull"
+fi
+
+## main processing loop
+for i in "${COMMITS[@]}"
+  do
+    LOG "INFO" "::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::";
+    LOG "INFO" "Processing commit          : $i"
+    LOG "INFO" "::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::";
+
+    $GIT_COMMAND cherry-pick "$i" -x
+    if [ $? -eq 0 ]; then
+      LOG "INFO" "Cherry pick of $i completed successfully."
+      continue;
+    fi
+    LOG "ERROR" "Cherry pick encountered an error."
+    if [[ "$AUTO_MODE" ]]; then
+      LOG ERROR "Aborting cherry-pick and exiting. Please handle this manually"
+      $GIT_COMMAND cherry-pick --abort
+      exit 2
+    fi
+    if yesno "Do you want me to open mergetool? (y/n) "; then
+      $GIT_COMMAND mergetool
+    fi
+    if yesno "Have you resolved the merge conflict? (y/n) "; then
```

Comment:
I think this can read the exit status of merge tool to know if it was resolved or not?

Reply:
Yea, perhaps, but an explicit prompt gives an explicit consent.. Can put it as future improvement